### PR TITLE
use short path name when running quarto

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/11779.

### Approach

Use a short path name when trying to run Quarto.

### Automated Tests

N/A (Windows only)

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/11779.

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests

### CODE REVIEW

**Review by jgutman — APPROVED** [review]
@jjallaire just created a new issue for this which got assigned to me. This seems like a reasonable solution to me, although it's odd that we don't have this issue with other executables, which does seem to suggest this could perhaps be addressed on the Quarto side, but I think addressing it this way is sufficient for now to make sure it gets into this release

**Comment by jgutman** [comment]
Also, we will soon have the ability to run automated tests on Windows, and this is a good candidate. Not sure of the ETA on that but I will put this on @jonvanausdeln's radar

File: src/cpp/core/FileUtils.cpp [hunk 1]
```diff
@@ -65,6 +65,29 @@ bool copySourceFile(const FilePath& sourceDir,
...
+
+   std::vector<WCHAR> buffer(length, 0);
+   DWORD result = ::GetShortPathNameW(widePath.c_str(), &buffer[0], length);
+   if (result == 0)
+      return path;
+
+   return string_utils::wideToUtf8(std::wstring(&buffer[0]));
+
+#else
+   return string;
```
Comment (jgutman) [inline src/cpp/core/FileUtils.cpp:86]: This should have been return `path`; fixed on main post-merge

Comment (kevinushey) [reply src/cpp/core/FileUtils.cpp:86]: Sorry about that! Thanks for fixing that up.

**Pull request merged**